### PR TITLE
Update SET docs for timestamping

### PR DESCRIPTION
**Pull request by haydentherapper**
SETs serve two purposes: The promise of inclusion, and a signed
timestamp from Rekor. We've moved towards requiring inclusion proofs,
but SETs are still needed if no TSA timestamp is provided. This doc
update clarifies that.

Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

### CODE REVIEW

**Comment by haydentherapper** [comment]
DNM til https://github.com/sigstore/protobuf-specs/pull/99 is in.

**Comment by haydentherapper** [comment]
Just look at the last commit for now

File: protos/sigstore_rekor.proto [hunk 1]
```diff
@@ -106,7 +106,7 @@ message TransparencyLogEntry {
         InclusionPromise inclusion_promise = 5;
         // The inclusion proof can be used for offline or online verification
         // that the entry was appended to the log, and that the log has not been
-        // altered.
+        // altered. Also may be used as a signed timestamp.
```
Comment (woodruffw) [inline protos/sigstore_rekor.proto:109]: Possibly a misunderstanding on my part, but shouldn't this comment be on `inclusion_promise`, not `inclusion_proof`? Or does `inclusion_proof` also serve as a signed timestamp?

Comment (haydentherapper) [reply protos/sigstore_rekor.proto:109]: Whoops, good catch, moved to `inclusion_promise`.

File: protos/sigstore_bundle.proto [hunk 2]
```diff
@@ -54,16 +54,16 @@ message VerificationMaterial {
...
         // Client verification libraries MAY provide an option to support v0.1
         // bundles for backwards compatibility, which may contain an inclusion
         // promise and not an inclusion proof. In this case, the client MUST
         // validate the promise.
         // Verifiers SHOULD NOT allow v0.1 bundles if they're used in an
         // ecosystem which never produced them.
         repeated dev.sigstore.rekor.v1.TransparencyLogEntry tlog_entries = 3;
-        // Timestamp verification data, over the artifact's signature.
+        // Timestamp may also come from
+        //tlog_entries.inclusion_promise.signed_entry_timestamp.
```
Comment (woodruffw) [inline protos/sigstore_bundle.proto:66]: Formatting nit: 

```suggestion
        // tlog_entries.inclusion_promise.signed_entry_timestamp.
```

**Review by woodruffw — APPROVED** [review]

**Pull request merged**